server/src/main/kotlin/io/spine/examples/pingh/server/Application.kt

```diff
@@ -108,11 +139,52 @@ internal class PinghApplication {
             .atPort(pinghPort)
             .add(
                 newSessionsContext(
-                    RemoteGitHubAuthentication(gitHubApp, httpEngine),
+                    RemoteGitHubAuthentication(gitHubApp(), httpEngine),
                     RemoteGitHubUsers(httpEngine)
                 )
             )
             .add(newMentionsContext(RemoteGitHubSearch(httpEngine)))
             .build()
     }
 }
+
+/**
+ * Returns the GitHub App secrets required to make authentication requests
+ * on behalf of the App.
+ *
+ * In [Production] mode, obtains secrets from the Secret Manager.
+ * In non-production mode, loads secrets from resource folder.
+ */
+private fun gitHubApp(): GitHubApp =
+    if (Environment.instance().`is`(Production::class.java)) {
+        GitHubApp::class.of(
+            ClientId::class.of(Secret.named("github_client_id")),
+            ClientSecret::class.of(Secret.named("github_client_secret"))
+        )
+    } else {
+        loadGitHubAppSecrets()
+    }
+
+/**
+ * Loads GitHub application secrets from resource folder.
+ */
+private fun loadGitHubAppSecrets(): GitHubApp {
+    val properties = Properties()
+    val path = "/local/config/server.properties"
+    Application::class.java.getResourceAsStream(path).use {
+        properties.load(it)
+    }
+    val messageFormat = "For running Pingh server locally the \"%s\" must be provided " +
```

Comment:
`the \"%s\" must be provided ` is missing the noun. _What_ it is that is missing?